packages/mint-components/src/components/sqm-coupon-code/sqm-coupon-code.feature

```diff
@@ -0,0 +1,137 @@
+@author:truman
+@owner:truman
+Feature: Coupon Code
+
+  The coupon code component is a box that allows users to see and copy their coupon code for a given program
+
+  @motivating
+  Scenario: A Users coupon code can be copied to their clipboard
+    Given tooltipText is "hello tooltip"
+    When the component renders
+    Then there is a textbox with the user's coupon code
+    When the clipboard icon is clicked
+    Then the code is copied to clipboard
+    And a tooltip appears for ~1 second
+
+  @motivating
+  Scenario: Component only fetches codes from current program
+    Given a user has multiple rewards from various programs
+    When the component is loaded
+    Then it fetches the user's rewards from only the program associated with the widget
+
+  @motivating
+  Scenario: Component only looks at fueltank rewards
+    Given a user has multiple rewards of different types
+    When the component is loaded
+    Then it filters the user's reward to only return FUELTANK rewards
+
+
+  @motivating
+  Scenario: The first (i.e. most recently received) reward is the reward that is displayed
+    Given a user has at least one fueltank reward
+    And the fueltank reward is available
+    Then the coupon code component shows the first reward returned
+    And this reward is the most recently received reward by the user
+
+  @motivating
+  Scenario Outline: Coupon code has multiple states depending on reward status
+    Given a user has at least one fueltank reward
+    And the reward has most recent status <status>
+    And the dateScheduledFor field is <dateScheduledFor>
+    Then the component's status is set to <componentStatus>
+    And the <errorMessageTextProp> is displayed under the input field
+
+    Examples:
+      | status    | dateScheduledFor | componentStatus | errorMessageTextProp     |
+      | AVAILABLE | null             | AVAILABLE       | N/A                      |
+      | EXPIRED   | null             | EXPIRED         | expiredErrorMessage      |
+      | REDEEMED  | null             | REDEEMED        | redeemedMessage          |
+      | CANCELLED | null             | CANCELLED       | cancelledErrorMessage    |
+      | PENDING   | null             | EMPTY_TANK      | fullfillmentErrorMessage |
+      | PENDING   | 123412341234     | PENDING         | pendingErrorMessage      |
+      | null      | null             | ERROR           | genericErrorMessage      |
+
+  Scenario Outline: Coupon code's error message text props are grouped
+    Given a user is viewing the coupon code component
```

Comment:
Who is the user in this case? What is this grouping about? Please add more details to this spec.